src/main/java/io/vertx/ext/sync/Sync.java

```diff
@@ -0,0 +1,171 @@
+   * @param <T>  the type of the result
+   * @return  the result
+   */
+  @Suspendable
+  public static <T> T syncResult(Consumer<Handler<AsyncResult<T>>> consumer) {
+    try {
+      return new AsyncAdaptor<T>() {
+        @Override
+        protected void requestAsync() {
+          try {
+            consumer.accept(this);
+          } catch (Exception e) {
+            throw new VertxException(e);
+          }
+        }
+      }.run();
+    } catch (Throwable t) {
+      throw new VertxException(t);
+    }
+  }
+
+  /**
+   * Receive a single event from a handler synchronously.
+   * The fiber will be blocked until the event occurs. No kernel thread is blocked.
+   *
+   * @param consumer  this should encapsulate the setting of the handler to receive the event. The handler is passed to it.
+   * @param <T>  the type of the event
+   * @return  the event
+   */
+  @Suspendable
+  public static <T> T syncEvent(Consumer<Handler<T>> consumer) {
+    try {
+      return new HandlerAdaptor<T>() {
+        @Override
+        protected void requestAsync() {
+          try {
+            consumer.accept(this);
+          } catch (Exception e) {
+            throw new VertxException(e);
+          }
+        }
+      }.run();
+    } catch (Throwable t) {
+      throw new VertxException(t);
+    }
+  }
+
+  /**
+   * Convert a standard handler to a handler which runs on a fiber. This is necessary if you want to do fiber blocking
+   * synchronous operations in your handler.
+   *
+   * @param handler  the standard handler
+   * @param <T>  the event type of the handler
+   * @return  a wrapped handler that runs the handler on a fiber
+   */
+  @Suspendable
+  public static <T> Handler<T> fiberHandler(Handler<T> handler) {
+    return p -> runOnFiber(getContextScheduler(), () -> handler.handle(p));
+  }
+
+  /**
+   * Create an adaptor that converts a stream of events from a handler into a receiver which allows the events to be
+   * received synchronously.
+   *
+   * @param <T>  the type of the event
+   * @return  the adaptor
+   */
+  @Suspendable
+  public static <T> HandlerReceiverAdaptor<T> streamAdaptor() {
+    return new HandlerReceiverAdaptorImpl<>(getContextScheduler());
+  }
+
+  /**
+   * Like {@link #streamAdaptor()} but using the specified Quasar `Channel` instance. This is useful if you want to
+   * fine-tune the behaviour of the adaptor.
+   *
+   * @param channel  the Quasar channel
+   * @param <T>  the type of the event
+   * @return  the adaptor
+   */
+  @Suspendable
+  public static <T> HandlerReceiverAdaptor<T> streamAdaptor(Channel<T> channel) {
+    return new HandlerReceiverAdaptorImpl<>(getContextScheduler(), channel);
+  }
+
+  /**
+   * Run the specified action on a fiber
+   *
+   * @param fiberScheduler  the schedulre to use
+   * @param runner  the action
+   */
+  @Suspendable
+  public static void runOnFiber(FiberScheduler fiberScheduler, Runnable runner) {
```

Comment:
Why is this method necessary? Also, this functionality is already found in [`co.paralleluniverse.fibers.FiberUtil`](https://github.com/puniverse/quasar/blob/master/quasar-core/src/main/java/co/paralleluniverse/fibers/FiberUtil.java).